rand_distr/src/gumbel.rs

```diff
@@ -0,0 +1,152 @@
+// Copyright 2018 Developers of the Rand project.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// https://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+//! The Gumbel distribution.
+
+use crate::{Distribution, OpenClosed01};
+use core::fmt;
+use num_traits::Float;
+use rand::Rng;
+
+/// Samples floating-point numbers according to the Gumbel distribution
+///
+/// # Example
+/// ```
+/// use rand::prelude::*;
+/// use rand_distr::Gumbel;
+///
+/// let val: f64 = thread_rng().sample(Gumbel::new(0.0, 1.0).unwrap());
+/// println!("{}", val);
+/// ```
+#[derive(Clone, Copy, Debug)]
+#[cfg_attr(feature = "serde1", derive(serde::Serialize, serde::Deserialize))]
+pub struct Gumbel<F>
+where
+    F: Float,
+    OpenClosed01: Distribution<F>,
+{
+    location: F,
+    scale: F,
+}
+
+/// Error type returned from `Gumbel::new`.
+#[derive(Clone, Copy, Debug, PartialEq, Eq)]
+pub enum Error {
+    /// location is infinite or NaN
+    LocationNotValid,
+    /// scale is not finite positive number
+    ScaleNotValid,
+}
+
+impl fmt::Display for Error {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.write_str(match self {
+            Error::ScaleNotValid => "scale is not positive and finite in Gumbel distribution",
+            Error::LocationNotValid => "location is not finite in Gumbel distribution",
```

Comment:
```suggestion
            Error::LocationNotValid => "location is not positive in Gumbel distribution",
```

Reply:
For this one I chose to use "finite" to mean "non-NaN and finite" (I believe the location actually can be negative even though the shape looks like that of a positive distribution). Would it be better  to say "location not real-valued" or something similar?

Reply:
You're right, this does not make sense, sorry. I somehow confused it with the scale.